libvast/src/bitmap.cpp

```diff
@@ -57,6 +89,60 @@ bool operator==(const bitmap& x, const bitmap& y) {
   return x.bitmap_ == y.bitmap_;
 }
 
+auto pack(flatbuffers::FlatBufferBuilder& builder, const bitmap& from)
+  -> flatbuffers::Offset<fbs::Bitmap> {
+  auto f = detail::overload{
+    [&](const ewah_bitmap& ewah) {
+      const auto ewah_offset = pack(builder, ewah).Union();
+      return fbs::CreateBitmap(builder, fbs::bitmap::Bitmap::ewah,
+                               ewah_offset.Union());
+    },
+    [&](const null_bitmap& null) {
+      const auto null_offset = pack(builder, null).Union();
+      return fbs::CreateBitmap(builder, fbs::bitmap::Bitmap::null,
+                               null_offset.Union());
+    },
+    [&](const wah_bitmap& wah) {
+      const auto wah_offset = pack(builder, wah).Union();
+      return fbs::CreateBitmap(builder, fbs::bitmap::Bitmap::wah,
+                               wah_offset.Union());
+    },
+  };
+  return caf::visit(f, from.bitmap_);
+}
+
+auto unpack(const fbs::Bitmap& from, bitmap& to) -> caf::error {
+  switch (from.bitmap_type()) {
+    case fbs::bitmap::Bitmap::NONE:
+      return caf::make_error(ec::logic_error, "invalid vast.fbs.Bitmap type");
+    case fbs::bitmap::Bitmap::ewah: {
+      const auto& from_ewah = *from.bitmap_as_ewah();
+      auto to_ewah = ewah_bitmap{};
+      if (auto err = unpack(from_ewah, to_ewah))
+        return err;
+      to.bitmap_ = std::move(to_ewah);
+      return caf::none;
```

Comment:
I think it should be possible to define a lambda that offers an API like:
```c++
return do_unpack(*from.bitmap_as_ewah(), ewah_bitmap{});
```
And use that for every case.

Reply:
Good idea, I've pushed that in a separate commit.